pennylane/io.py

```diff
@@ -414,15 +414,36 @@ def from_qasm(quantum_circuit: str, measurements=False):
     """Loads quantum circuits from a QASM string using the converter in the
     PennyLane-Qiskit plugin.
 
+    Args:
+        quantum_circuit (str): a QASM string containing a valid quantum circuit
+        measurements (None | MeasurementProcess | list[MeasurementProcess]): an optional PennyLane
+            measurement or list of PennyLane measurements that overrides any terminal measurements
+            that may be present in the input circuit. If set to ``None``, existing measurements
+            in the input circuit will be used.
+
+    Returns:
+        function: the PennyLane template created based on the QASM string
+
+    .. warning::
+
+        The current default behaviour of removing measurements in the QASM code is deprecated
+        and will be changed in a future release. Starting in version ``0.38``, ``from_qasm``
+        will keep the measurements from the QASM code by default. To remove all measurements,
+        set ``measurements=[]`` which overrides the existing measurements with an empty list.
+
     **Example:**
 
     .. code-block:: python
 
         >>> hadamard_qasm = 'OPENQASM 2.0;' \\
         ...                 'include "qelib1.inc";' \\
         ...                 'qreg q[1];' \\
-        ...                 'h q[0];'
+        ...                 'creg c[1];' \\
+        ...                 'h q[0];' \\
+        ...                 'measure q -> c;'
         >>> my_circuit = qml.from_qasm(hadamard_qasm)
+        >>> my_circuit()
+        (MeasurementValue(wires=[0]),)
```

Comment:
Yes. I updated the example to include a measurement in the QASM, so this should be the up-to-date version

Reply:
Yes. I added a measurement to the QASM code and that's why it looks like that.

Reply:
But the default behaviour should be to overwrite those with an empty list (no measurements) until next release. Since the default is `measurements=False`. 

I only get the output you include above if I run `my_circuit = qml.from_qasm(hadamard_qasm, measurements=None)`.